xswap/preprocessing.py

```diff
@@ -0,0 +1,106 @@
+import pathlib as _pathlib
+
+
+def load_str_edges(filename, node_delim=',', edge_delim='\n'):
+    """
+    Load edges from file into memory. Store edges as a list and store each edge
+    as Tuple[str, str]. Used to load edges for preprocessing.
+    """
+    file_path = _pathlib.Path(filename)
+    if not file_path.is_file():
+        raise FileNotFoundError
+
+    with open(file_path, 'r') as f:
+        graph_string = f.read()
+    str_edges = [
+        edge.split(node_delim) for edge in graph_string.split(edge_delim) if edge != ''
```

Comment:
Would it not make sense to use Python's CSV module here, which could deal with things like quoting.